.github/workflows/macos.yml

```
@@ -71,25 +71,31 @@ jobs:
 
       - run: make
 
+      - name: Set test options for skipped tests
+        run: |
+          set -ex
+          TESTS="$(echo "${{ matrix.skipped_tests }}" | sed 's| |$$/ -n!/|g;s|^|-n!/|;s|$|$$/|')"
+          echo "TESTS=${TESTS}" >> $GITHUB_ENV
+        if: ${{ matrix.test_task == 'check' && matrix.skipped_tests }}
+
       - name: make ${{ matrix.test_task }}
         run: |
-          make -s ${{ matrix.test_task }} ${TESTS:+TESTS=`echo "$TESTS" | sed 's| |$$/ -n!/|g;s|^|-n!/|;s|$|$$/|'`}
+          make -s ${{ matrix.test_task }} ${TESTS:+TESTS="$(echo "$TESTS")"}
         timeout-minutes: 60
         env:
           RUBY_TESTOPTS: '-q --tty=no'
-          TESTS: ${{ matrix.test_task == 'check' && matrix.skipped_tests || '' }}
           TEST_BUNDLED_GEMS_ALLOW_FAILURES: 'minitest,test-unit,debug,bigdecimal,drb,typeprof'
           PRECHECK_BUNDLED_GEMS: 'no'
 
       - name: make skipped tests
         run: |
-          make -s test-all TESTS=`echo "$TESTS" | sed 's| |$$/ -n/|g;s|^|-n/|;s|$|$$/|'`
+          make -s test-all TESTS="$(echo "$TESTS" | sed 's| |$$/ -n/|g;s|^|-n/|;s|$|$$/|')"
         env:
           GNUMAKEFLAGS: ''
           RUBY_TESTOPTS: '-v --tty=no'
           TESTS: ${{ matrix.skipped_tests }}
```

How about reusing substituted `$TESTS`?

```suggestion
          make -s test-all TESTS="${TESTS//-n!\//-n/}"
        env:
          GNUMAKEFLAGS: ''
          RUBY_TESTOPTS: '-v --tty=no'
```

---

It seems that this change makes the code simpler.
I tried your suggestion on my local by using the following testing script. It looks working. I will change my code.

```
$ cat test.sh 
#!/bin/bash

set -ex

TESTS="-n!/TestGem#test_.*\$$/ -n!/TestMkmf.*\$$/"
TESTS="${TESTS//-n!\//-n/}"
echo $TESTS
```

```
$ bash test.sh 
+ TESTS='-n!/TestGem#test_.*$$/ -n!/TestMkmf.*$$/'
+ TESTS='-n/TestGem#test_.*$$/ -n/TestMkmf.*$$/'
+ echo '-n/TestGem#test_.*$$/' '-n/TestMkmf.*$$/'
-n/TestGem#test_.*$$/ -n/TestMkmf.*$$/
```

---

I fixed it.